library/control/test/workflow_manager_test.rb

```diff
@@ -386,22 +386,59 @@
       subject.control_file(repo_id)
     end
 
-    it "returns nil if the extracted package does not contain installation.xml" do
-      expect(File).to receive(:exist?).with(/installation\.xml\z/).and_return(false)
-      expect(subject.control_file(repo_id)).to be nil
-    end
+    context "if downloading and extracting worked" do
+      before do
+        allow(Dir).to receive(:glob).with(/installation-products/).and_return product_files
+        allow(Dir).to receive(:glob).with(/system-roles/).and_return role_files
 
-    it "returns the installation.xml path if the extracted package contains it" do
-      expect(File).to receive(:exist?).with(/installation.xml\z/).and_return(true)
-      # the returned path contains "/installation.xml" at the end
-      expect(subject.control_file(repo_id)).to end_with("/installation.xml")
-    end
+        allow(File).to receive(:exist?) do |name|
+          product_files.include?(name) || role_files.include?(name)
+        end
+
+        allow(File).to receive(:exist?).with(/installation\.xml\z/).and_return installation_xml
+      end
+
+      context "if the package contains a control file in the system-roles dir" do
```

Comment:
Do you think that it worth it to add another scenario with files in both, `product_files` and `system_roles`?